### ISSUE TITLE
Release schedule for julia v0.7

### ISSUE THREAD

**Issue by jpfairbanks** [JuliaGraphs/LightGraphs.jl#793]
We should come up with a schedule for a release for julia version 0.7 and version 1.0.

I would like to learn from the previous release schedule. 

- [x] Feature Freeze
- [x] Tag a last version supporting 0.6
- [x] Master branch is 0.7+ (bump REQUIRE)
- [x] Fix all the bugs on deprecations on 0.7 on master
- [x] Tag a first version supporting 0.7

@sbromberger what do you think? Should we make a project or milestone?

**Comment by sbromberger** [comment]
Good idea to plan this in advance.

1. I think our feature freeze should be December 15 (same as for Julia).
2. yes. This has worked very well in the past. I think this can be done right after feature freeze.
3. Yep. Right after 2.
4. Yep, right after 3.
5. Yep, right after 4.

Having gone through projects in GSOC, I'm not quite sure I like them. - but if you want to give it a try, I'm willing to give it another go.

**Comment by sbromberger** [comment]
On second thought, I'm not sure we need to formalize a feature freeze. I mean, hypothetically, we can do it right now and release a final 0.6 version today, moving directly into fixing 0.7 deprecation warnings and errors.

I'm not advocating this, but I don't know that setting a firm date is really necessary. It's not like we have a backload of new features that need implementing. Thoughts?

**Comment by sbromberger** [comment]
Also, because this process is fairly short and simple, I'll just rewrite your comment to be a to-do list and we'll use this for tracking.

**Comment by jpfairbanks** [comment]
Is now a good time to do a feature freeze? We have only three PRs outstanding.

**Comment by sbromberger** [comment]
Let's set it for mid-February to give @adamglos92 a chance to get his random graph generators in.

**Comment by sbromberger** [comment]
OK, we're now finished with "Master branch is 0.7+ (bump REQUIRE)".

**Comment by jpfairbanks** [comment]
Woohoo!

**Comment by jpfairbanks** [comment]
We are getting close to releasing v0.13 

I get the following depwarns locally on julia v"0.7.0-DEV.4390"

```julia
julia> Pkg.test("LightGraphs")
[ Info: Testing LightGraphs
┌ Warning: `Array{T, 1}(m::Int) where T` is deprecated, use `Array{T, 1}(uninitialized, m)` instead.
│   caller = Type at buffer.jl:30 [inlined]
└ @ Core buffer.jl:30
┌ Warning: `Array{T, 1}(m::Int) where T` is deprecated, use `Array{T, 1}(uninitialized, m)` instead.
│   caller = Type at buffer.jl:30 [inlined]
└ @ Core buffer.jl:30
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `nb_available` is deprecated, use `bytesavailable` instead.
│   caller = readdata!(::IOStream, ::TranscodingStreams.Buffer) at stream.jl:615
└ @ TranscodingStreams stream.jl:615
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `nb_available` is deprecated, use `bytesavailable` instead.
│   caller = readdata!(::IOStream, ::TranscodingStreams.Buffer) at stream.jl:618
└ @ TranscodingStreams stream.jl:618
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
^[┌ Warning: `Array{T, 1}(m::Int) where T` is deprecated, use `Array{T, 1}(uninitialized, m)` instead.
│   caller = Type at buffer.jl:30 [inlined]
└ @ Core buffer.jl:30
┌ Warning: `Array{T, 1}(m::Int) where T` is deprecated, use `Array{T, 1}(uninitialized, m)` instead.
│   caller = Type at buffer.jl:30 [inlined]
└ @ Core buffer.jl:30
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
┌ Warning: `unsafe_copy!(dest, args...)` is deprecated, use `unsafe_copyto!(dest, args...)` instead.
│   caller = copydata! at buffer.jl:192 [inlined]
└ @ Core buffer.jl:192
┌ Warning: `endof(a)` is deprecated, use `lastindex(a)` instead.
│   caller = marginsize at buffer.jl:59 [inlined]
└ @ Core buffer.jl:59
```

**Comment by sbromberger** [comment]
Yup. None of those are ones we can fix in LightGraphs. I think we’re ready to tag 0.13. Thoughts?

**Comment by sbromberger** [comment]
we're now finished with "Fix all the bugs on deprecations on 0.7 on master".

**Comment by sbromberger** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/13546
